tests/v2/validation/registries/registry_test.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	corralRegistryRancherImageName = "registryrancher"
+	corralRegistryRancherName      = "rancherglobalregistry"
+	corralAuthDisabledImageName    = "registryauthdisabled"
+	corralAuthDisabledName         = "downstreamregistrydisabled"
+	corralAuthEnabledImageName     = "registryauthenabled"
+	corralAuthEnabledName          = "downstreamregistryenabled"
+)
+
+var (
+	adminPassword = os.Getenv("ADMIN_PASSWORD")
+)
+
+type RegistryTestSuite struct {
+	suite.Suite
+	session                        *session.Session
+	podListClusterAuth             *v1.SteveCollection
+	podListClusterNoAuth           *v1.SteveCollection
+	podListClusterLocal            *v1.SteveCollection
+	clusterAuthRegistryHost        string
+	clusterNoAuthRegistryHost      string
+	localClusterGlobalRegistryHost string
+}
+
+func (rt *RegistryTestSuite) TearDownSuite() {
+	ranchersetup.RancherCleanup(rt.session)
+	corralList, err := corral.ListCorral()
+	require.NoError(rt.T(), err)
+	err = corralext.CleanUpCorrals(corralList)
+	require.NoError(rt.T(), err)
+	rt.session.Cleanup()
+}
+
+func (rt *RegistryTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rt.session = testSession
+
+	corralConfig := corral.CorralConfigurations()
+	err := corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(rt.T(), err)
+
+	configPackage := corral.CorralPackagesConfig()
+
+	path := configPackage.CorralPackageImages[corralRegistryRancherImageName]
+	_, err = corral.CreateCorral(testSession, corralRegistryRancherName, path, true, configPackage.Cleanup)
+	require.NoError(rt.T(), err)
+
+	password, err := corral.GetCorralEnvVar(corralRegistryRancherName, "bootstrap_password")
+	require.NoError(rt.T(), err)
+	globalRegistryFqdn, err := corral.GetCorralEnvVar(corralRegistryRancherName, "registry_fqdn")
+	require.NoError(rt.T(), err)
+
+	rancherConfig := new(rancher.Config)
+	config.LoadConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	token, err := ranchersetup.CreateAdminToken(password, rancherConfig)
+	require.NoError(rt.T(), err)
+
+	rancherConfig.AdminToken = token
+	config.UpdateConfig(rancher.ConfigurationFileKey, rancherConfig)
+
+	session := session.NewSession()
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+	kubernetesVersions := clustersConfig.RKE1KubernetesVersions
+	cnis := clustersConfig.CNIs
+	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient(token, session)
+	require.NoError(rt.T(), err)
+
+	err = ranchersetup.PostRancherInstall(client, adminPassword)
+	require.NoError(rt.T(), err)
+
+	corralConfig = corral.CorralConfigurations()
```

Comment:
We should encapsulate the corral registry creations with a if statement that checks a flag to see if it needs to be created. That way when we run our validations, we're not always creating registries. I think it makes more sense to have the cluster provisions be apart of the tests validations below, since that is a part of the validations. I just want to avoid having to create private registries every time we run this test especially if they already exist. This will make debugging and manual validations extremely hard, some thing we occasionally deal with the python tests.

Reply:
I agree, I can see this would enable us to re-run only the validations without creating private registries, which I believe is the most time spent. @izaac would it be helpful in case we get a bug and want to rerun or reproduce it? 

So wrapping this with an `environmentflag` might be a really good approach to avoid registry creations